### DISCUSSION TITLE
Looking for sample QR Codes

### DISCUSSION THREAD

**Discussion by rkreutzer** [decentralized-identity/veramo#357] [Q&A]
I'm wanting to test the URL message handler in my app with a variety of QR Codes to simulate all the possible options. Does anyone have a good reference for this? If anyone has pieces of this, I'm willing to consolidate and share a doc.

**Comment by mirceanis** [comment]
How about this one:

http://localhost?c_i=eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJ2YyI6eyJjcmVkZW50aWFsU3ViamVjdCI6eyJuYW1lIjoiQWxpY2UifSwiQGNvbnRleHQiOlsiaHR0cHM6Ly93d3cudzMub3JnLzIwMTgvY3JlZGVudGlhbHMvdjEiXSwidHlwZSI6WyJWZXJpZmlhYmxlQ3JlZGVudGlhbCIsIlByb2ZpbGUiXX0sInN1YiI6ImRpZDpldGhyOjB4ZTQyYjAwMTJiNDg3ZDYzNDI5NTRhNWYwYjY2NTRmNDE5YTdiNjQxMCIsIm5iZiI6MTYxMTkzOTY3MSwiaXNzIjoiZGlkOmV0aHI6MHhlNDJiMDAxMmI0ODdkNjM0Mjk1NGE1ZjBiNjY1NGY0MTlhN2I2NDEwIn0.xFUnRelHfDE5bkHS5929ThVMFhfETAJUZwpbSWaxTLn1KuaV_Nufk9NwNdsFoYsPoaxhETbx41SVYhyiIePjLA

It's a self-signed Profile credential containing a `name` claim.
If this is not what you are looking for, can you be more specific?

Are you using `@veramo/url-handler`, or have you created your own?

**Reply by vongohren** [reply]
Just a question to this? There is no did connection involved here? And what are some ways of presenting this in a QR code when they start to become bigger? Is this a discussion for another place maybe 😅?

Is that when you start adding a relay endpoint so that when connected the can request the VC from an endpoint?

**Reply by rkreutzer** [reply]
Thanks for the JWT. That scans nicely, and yes, I am using the url-handler. The issue now is with saving it, as apparently the subject of the VC must match to one of the DIDs in my database. That makes sense, so I'll need to revise how I test. I'm writing a routine that will accept a VC JSON string, change the subject to one of my DIDs, then create a JWT and URL from that.  If the VC is signed, then the signature verification will fail, which is fine. Do you have additional sample VCs that I could use?

**Reply by rkreutzer** [reply]
With regard to large QR Codes, I'm creating a short URL that redirects to the long JWT URL. For my app, it will be when a user generates a Verifiable Presentation. I've also seen others use this approach. I don't know enough about endpoints to comment on that. 

The current url-handler doesn't support URL redirects, so I'll contribute back code for this soon.

**Comment by vongohren** [comment]
But short urls are a bottleneck for centralization so it is an easy fix for
this, but I really want to know if anyone knows anything about how it can
work.
Or is it centralization when you are already talking to the issuer via
connection, and receiving just a short url, should not matter :thinking:
My suggestion was not very decentralized either
